packages/core/docs/integrate/integration-guide.md

```diff
@@ -340,78 +255,108 @@ const factoryParams: UseUserFactoryParams = {
 
-const getItems = (cartResponse) => cartResponse.results.lineItems;
+```typescript
+// packages/api-client/src/index.server.ts
+import { getProduct } from './api/getProduct';
 
-const getItemPrice = (lineItem: LineItem): AgnosticPrice => ({
-  regular: lineItem.price.amount,
-  special: lineItem.price.discounted ? lineItem.price.discounted : 0
+// Unrelated code omitted
+
+const { createApiClient } = apiClientFactory<any, any>({
+  onCreate,
+  api: {
+    getProduct
+  }
 });
+```
 
-const getItemName = (lineItem: LineItem): string => lineItem.masterVariant.name;
+### Understand getters
 
-const getItemQty = (lineItem: LineItem): number => lineItem.quantity;
+Now that we can request and save API data, we need to display it to the user. Every integration can use and extend our base theme for that. However, accessing the data straight from integrations isn't possible because each uses unique data structures. To do that, we need getters.
 
-const cartGetters: CartGetters = {
-  getItems,
-  getItemPrice,
-  getItemName,
-  getItemQuantity
-}
-```
+Getters allow us to take raw responses from the API and map or extract data to a standard format, which we use in the template.
 
-```ts
-// composables/src/index.js
-import cartGetters from './getters';
+### Implement `productGetters`
 
-export { cartGetters }
-```
+Open `packages/composables/src/getters/productGetters.ts`. There is a bunch of functions returned in a single object. Each of them has at least one argument: either a raw response from the API (`products`) or a single item extracted from it (`product`).
+
+Although types of arguments are unknown and specific to your integration, return types are already defined and must match those defined in [ProductGetters interface](../core/api-reference/core.productgetters.html).
 
-## Creating a theme
+You need to implement all of these functions and, if necessary, add your own.
+
+:::tip What if data is not available?
+We recommend that you always add fallback value in getters. This will prevent errors during Server-Side Rendering and avoid some edge cases if data or nested properties are not available, e.g., still being retrieved from the backend.
+
+```typescript
+export const getProductName = (product: ProductVariant): string => {
+  // Return empty string if "product" object or "name" property are not available
+  return product?.name || '';
+};
+```
+:::
 
-In the default theme, components used to display or modify integration-specific data, like forms or checkout information are blank. This is because each integration might use different data formats and properties.
+## Create a theme
 
-You need to create few Vue components and JavaScript files:
+Some form or checkout components are blank in the default theme because they display or modify integration-specific data. For this reason, you need to create few Vue components and JavaScript files:
```

Comment:
```suggestion
Some forms or checkout components are blank in the default theme because they display or modify integration-specific data. For this reason, you need to create few Vue components and JavaScript files:
```
Plural, right?